subprojects/language-java/src/main/java/org/gradle/api/internal/tasks/compile/AnnotationProcessorDetector.java

```diff
@@ -113,7 +115,9 @@ public Boolean calculate(File file, FileType fileType) {
                         zipFile.close();
                     }
                 } catch (IOException e) {
-                    throw new UncheckedIOException("Could not read service definition from JAR " + file, e);
+                    // Using a warning, not a deprecation message, because we already show a deprecation message during
+                    // compile classpath snapshotting, but we still want to warn
+                    LOGGER.warn("Could not read service definition from JAR " + file);
```

Comment:
Compile classpath and annotation processor path can be different, so we cannot make the assumption from the comment. We should use the deprecation logger here too.